dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/RequestParamsValidatorTest.java

```diff
@@ -353,6 +366,88 @@ void shouldParseFilterWhenFilterHasMultipleOperatorWithFinalColon() throws BadRe
         filters);
   }
 
+  @Test
+  void shouldParseDataElementFilters() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters =
+        parseDataElementFilters(DE_1_UID + ":lt:20:gt:10," + DE_2_UID + ":like:foo");
+
+    assertEquals(
+        Map.of(
+            DE_1_UID,
+            List.of(
+                new QueryFilter(QueryOperator.LT, "20"), new QueryFilter(QueryOperator.GT, "10")),
+            DE_2_UID,
+            List.of(new QueryFilter(QueryOperator.LIKE, "foo"))),
+        filters);
+  }
+
+  @Test
+  void shouldParseDataElementFilterWhenOnlyUIDProvided() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters = parseDataElementFilters(DE_1_UID.getValue());
+
+    assertEquals(Map.of(DE_1_UID, List.of()), filters);
+  }
+
+  @Test
+  void shouldParseDataElementFilterWhenSingleUnaryOperator() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters = parseDataElementFilters(DE_1_UID + ":!null");
+
+    assertEquals(Map.of(DE_1_UID, List.of(new QueryFilter(QueryOperator.NNULL))), filters);
+  }
+
+  @Test
+  void shouldParseDataElementFilterWhenMultipleUnaryOperatorsCombined() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters = parseDataElementFilters(DE_1_UID + ":!null:null");
+
+    assertEquals(
+        Map.of(
+            DE_1_UID,
+            List.of(new QueryFilter(QueryOperator.NNULL), new QueryFilter(QueryOperator.NULL))),
+        filters);
+  }
+
+  @Test
+  void shouldParseDataElementFilterWhenUnaryAndBinaryOperatorsCombined()
+      throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters = parseDataElementFilters(DE_1_UID + ":null:gt:10");
+
+    assertEquals(
+        Map.of(
+            DE_1_UID,
+            List.of(new QueryFilter(QueryOperator.NULL), new QueryFilter(QueryOperator.GT, "10"))),
+        filters);
+  }
+
+  @Test
+  void shouldFailParsingDataElementFiltersWhenUnaryOperatorContainsValue() {
+    Exception exception =
+        assertThrows(
+            BadRequestException.class, () -> parseDataElementFilters(DE_1_UID + ":!null:value"));
+    assertEquals(
+        "Query item or filter is invalid: " + DE_1_UID + ":!null:value", exception.getMessage());
+  }
+
+  @Test
+  void
+      shouldFailParsingDataElementFiltersWhenUnaryAndBinaryOperatorsCombinedAndUnaryContainsValue() {
+    Exception exception =
+        assertThrows(
+            BadRequestException.class,
+            () -> parseDataElementFilters(DE_1_UID + ":gt:10:null:value"));
+    assertEquals(
+        "Operator in filter can't be used with a value: " + DE_1_UID + ":gt:10:null:value",
```

Comment:
Would be great if we can tell the user which operator is not allowed to have a value = unary. Same in above case.